### PR TITLE
#13

### LINKED ISSUE #13: Trainer interfaces
**Acceptance Criteria**:
- The following interfaces exist:
  - MirrorDataset: inherits IterableDataset. `__iter__` is typed to return a token tensor and an attention mask tensor, both of shape (T).
  - MirrorModel: inherits nn.Module.
    - `training_step` takes in a batch of tokens (B, T) and a batch of attention masks (B, T), and returns a loss
    - `configure_optimizers` is typed to return an optimizer
- The above interfaces have a single, placeholder implementation each. These should be *very* basic, just returning some random value of the right shapes. Full implementations will come in later tickets.
- Trainer class is created:
  - `fit` takes in a `MirrorDataset` and a `MirrorModel`, and returns nothing. For now, just `pass`, implementation will come later.
  - is also has a `fabric` property that is a `Fabric` instance
- A trainer is created in the main method. If the `fit` subcommand is used, the `fit` method is called with a placeholder model and dataset.

### PR DESCRIPTION
**Pull request by Plyb**
closes #13 

How I tested: not much functionality here, but I did run the script to make sure all the imports and such were working.

### CODE REVIEW

**Review by naataaniitsosie — COMMENTED** [review]
What's our approach for testing? Do we have any unit tests? I know you said we can manually test, but do we have any unit tests to ensure we don't break our classes and things like that?

**Comment by Plyb** [comment]
> What's our approach for testing? Do we have any unit tests? I know you said we can manually test, but do we have any unit tests to ensure we don't break our classes and things like that?

Haven't gotten that far yet. In my personal opinion, things like unit tests usually end up being just a form of code duplication if they aren't handled carefully. Integration tests are better, but more expensive to run (especially in a compute constrained domain like ML). I can bring up testing in the next MIRROR team meeting.
